lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Context/ContextRuntime.cs

```diff
@@ -296,16 +296,25 @@ public Thread StartTask(IConfiguration taskConfiguration)
 
                 var taskInjector = _contextInjector.ForkInjector(taskConfiguration);
 
-                var taskRuntime = taskInjector.GetInstance<TaskRuntime>();
-
                 try
                 {
+                    var taskRuntime = taskInjector.GetInstance<TaskRuntime>();
                     _task = Optional<TaskRuntime>.Of(taskRuntime);
                     return taskRuntime.RunTask();
                 }
                 catch (Exception e)
```

Comment:
Should you distinct injection exception that may potentially throw from taskInjector.GetInstance<TaskRuntime>()?